src/storage/src/source/persist_source.rs

```diff
@@ -182,11 +177,11 @@ where
                             let session_cap = cap_set.delayed(&current_ts);
                             let mut session = output.session(&session_cap);
 
+                            // Give the batch to a random worker.
+                            let worker_idx = usize::cast_from(Instant::now().hashed()) % peers;
                             let progress = batch.generate_progress();
-                            session.give((i, batch.get_droppable_batch()));
+                            session.give((worker_idx, batch.get_droppable_batch()));
```

Comment:
`worker_idx` is only used to perform the round robbin; you might be able to move `Instant::now().hashed()` down [here](https://github.com/MaterializeInc/materialize/pull/14452/files#diff-9808a8978ab8134e00035d965c2e44efa92a2fbb26cfa8bb4420a1b5358d0523R224) and get rid of the dog and pony show to strictly round robin.

Reply:
I personally like the explicit-ness of "batch distribution" making this decision and everything after that being deterministic